src/core/server/elasticsearch/client/cluster_client.ts

```diff
@@ -95,12 +95,15 @@ export class ClusterClient implements ICustomClusterClient {
   private getScopedHeaders(request: ScopeableRequest): Headers {
     let scopedHeaders: Headers;
     if (isRealRequest(request)) {
-      const authHeaders = this.getAuthHeaders(request);
       const requestHeaders = ensureRawRequest(request).headers;
-      scopedHeaders = filterHeaders(
-        { ...requestHeaders, ...authHeaders },
-        this.config.requestHeadersWhitelist
-      );
+      const requestIdHeaders =
+        request instanceof KibanaRequest ? { 'x-opaque-id': request.id } : {};
```

Comment:
NIT: maybe export and use (here + in `src/core/server/elasticsearch/legacy/cluster_client.ts`)

https://github.com/elastic/kibana/blob/10cc600d5cdc9239ce174683b9fee6c3b22c8f4c/src/core/server/http/router/request.ts#L273-L275

for consistency